### ISSUE TITLE
HTTP webtriggered functions don't seem to run in parallel (can it be forced)

### ISSUE THREAD

**Issue by moab360** [Azure/azure-functions-host#1401]
I'm working on moving as much of my server workloads over to functions as I can.  There is one user facing http GET in particular that requires parsing up a bunch of images and returning some data.  On the web server, this can be done pretty quickly (parallel.for loop to the rescue) even when multiple users call the service concurrently.  But it requires I have a pretty beefy server to do the work

When moved over to a function, I broke it up so that one function ( F(a) ) spawns multiple functions to do the image work ( F(b) ), then return back to F(a) (and then the customer).  

The issue that I'm noticing is that when two customers call F(a) nearly concurrently, the first call returns fairly quickly, but the second call takes the first calls time, plus the time to run the function for itself

Poking through the timings, this is what I'm seeing in my test cases
Call 1 to f(a)
2017-04-12T21:27:30.663 Function started (Id=d28fd004-6ba8-49d5-8990-4a9bb5249bc2)
2017-04-12T21:27:36.821 Function completed (Success, Id=d28fd004-6ba8-49d5-8990-4a9bb5249bc2)

Call 2 (initiated about 100ms after call 1)
2017-04-12T21:27:37.114 Function started (Id=389a6ccb-396c-4e85-8eb8-4d757b6c1bd5)
2017-04-12T21:27:43.136 Function completed (Success, Id=389a6ccb-396c-4e85-8eb8-4d757b6c1bd5)

I'd expect that Call 2, since started 100ms later, would have it's function start roughly 100ms later and run in parallel with the first function.  Adding more calls just continues this 

Here is what it looks like from the customers point of view
call 1
![image](https://cloud.githubusercontent.com/assets/15697848/24980441/15234632-1f8d-11e7-8126-053382db1b71.png)

call 2
![image](https://cloud.githubusercontent.com/assets/15697848/24980406/e7969b56-1f8c-11e7-8710-9ae06fe72b6d.png)

Is this expected and is it possible to force functions to run in parallel?

I should note that when F(a) starts multiple F(b), all F(b) run in parallel.

**Comment by christopheranderson** [comment]
This is a limitation of our current Node.js implementation that we'll try to reuse the Node.js process. You shouldn't use blocking Node.js code, if you can avoid it. We'll eventually scale out more processes.

We're working on a new execution environment for Node.js that will give us more flexibility to force this to independent processes. For now, you can't fix this short of forking the Node.js process yourself in your code, which is not trivial and not a good solution for us to give you. :(

**Comment by moab360** [comment]
Thanks for the reply.  To be clear, this is a C# HTTP triggered function, so I'm not sure if the Node.js execution environment applies here.

**Comment by kshyju** [comment]
I don’t believe this is still the case with .NET Function Apps today. Both .NET and Azure Functions have evolved significantly and generally offer much better performance than in the past. If you’re still encountering this issue, could you please share a minimal repro so we can take a closer look?
